File: 이것이 코딩 테스트다/Greedy/모험가 길드.py
#모험가 길드
N = int(input())
data = list(map(int,input().split()))

data.sort()
result, temp, count = 0, 0, 0 

for i in data :
  if i == 1 :
    result += 1
    continue
    
  else :
    count += 1
    if temp < i :
      temp = i
  
  if count == temp :
    count, temp = 0, 0
    result += 1

print(result)

File: 이것이 코딩 테스트다/Greedy/모험가 길드.java
import java.util.*;

public class Num6 {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		
		int[] arr = new int[n];
		for(int i = 0; i < n; i++) {
			arr[i] = sc.nextInt();
		}
		
		Arrays.sort(arr);
		int count = 0;
		int result = 0;
		
		for(int i = 0; i < n; i++) {
			count++;
			if(count == arr[i]) {
				result++;
				count = 0;
			}
		}
		
		System.out.println(result);
	}
}
